packages/sanity/src/core/studio/components/navbar/search/definitions/fields.ts

```diff
@@ -51,15 +72,30 @@ export function createFieldDefinitions(
   // Get supported filter field types that have corresponding filters defined
   const supportedFieldTypes = getSupportedFieldTypes(filterDefinitions)
 
-  // Recursively iterate through all documents and resolve objects
   return getDocumentFieldDefinitions(supportedFieldTypes, documentTypes, objectTypes)
 }
 
+export function createFieldDefinitionDictionary(
+  fieldDefinitions: SearchFieldDefinition[]
+): SearchFieldDefinitionDictionary {
+  return fieldDefinitions.reduce<SearchFieldDefinitionDictionary>((acc, val) => {
+    acc[val.id] = val
+    return acc
+  }, {})
+}
+
+export function generateFieldId(field: SearchFieldDefinition): string {
+  return Md5.hashStr(
```

Comment:
Don't have the full context here, but could this be simplified to `[field.documentTypes.join(','), field.fieldPath, field.filterName, field.type].join('-')`?

Reply:
Great callout – this hashing is completely unnecessary, it's not used for any comparison and the joined string is already unique by design. That's been changed!